### PR TITLE
enhance: Improve compatibility with SSR and React Native

### PR DESCRIPTION
**Pull request by ntucker**
### Motivation

CacheProvider is needed in all installs, so ensure it works in all compatible cases.

### Solution

- .native extension allows for RN specific version that uses <Text>
- Don't use <Suspense/> in React 16, 17 when doing SSR - as it will show fallback which will likely block entire app

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -112,11 +112,15 @@
   },
   "peerDependencies": {
     "@types/react": "^16.8.4 || ^17.0.0 || ^18.0.0-0",
-    "react": "^16.8.4 || ^17.0.0 || ^18.0.0-0"
+    "react": "^16.8.4 || ^17.0.0 || ^18.0.0-0",
+    "react-native": ">=0.59"
```
Comment (ljharb) [inline packages/core/package.json:116]: This is a breaking change for any package manager that doesn’t understand peerDependenciesMeta, and will force those react web consumers to install react-native. I very much doubt this is a desired approach.

I’d suggest a separate package for RN users, so non-RN users don’t have to interact with RN requirements.

File: packages/core/rollup.config.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ const dependencies = Object.keys(pkg.dependencies)
       !['@rest-hooks/use-enhanced-reducer', '@babel/runtime'].includes(dep),
   );
 
-const extensions = ['.js', '.ts', '.tsx', '.mjs', '.json', '.node'];
+const extensions = ['.js', '.ts', '.tsx', '.mjs', '.json', '.node', 'jsx'];
```
Comment (ljharb) [inline packages/core/rollup.config.js:18]: why does jsx omit the dot and the others include it?

File: packages/core/package.json (hunk 1 above)
Comment (ntucker) [reply packages/core/package.json:116]: "force those react web consumers to install react-native" - isn't the only version of npm that does this npm 7+, which does understand optional?

File: packages/core/rollup.config.js (hunk 2 above)
Comment (ntucker) [reply packages/core/rollup.config.js:18]: ack, good catch

File: packages/core/package.json (hunk 1 above)
Comment (ljharb) [reply packages/core/package.json:116]: By force, i mean that `npm ls` will exit nonzero, which is true in every version of npm. Making someone’s dep graph invalid is a problem even if it’s not exposed by default on npm 3-6.

Comment (ljharb) [reply packages/core/package.json:116]: Is there a reason it can't be compatible with RN without adding the peer dep?

Comment (ntucker) [reply packages/core/package.json:116]: I guess the node module resolution algorithm will still work even without marked peerdeps.

Comment (ntucker) [reply packages/core/package.json:116]: Only way .native will work is through metro so it's be really weird if RN wasn't found.

**Review by ljharb — APPROVED** [review]

**Pull request merged**